src/acl.c

```diff
@@ -934,6 +958,93 @@ static void ACLAddAllowedFirstArg(aclSelector *selector, unsigned long id, const
     selector->allowed_firstargs[id][items - 2] = sdsnew(sub);
     selector->allowed_firstargs[id][items - 1] = NULL;
 }
+/* Initialize intset for database permissions. */
+static intset *ACLInitDatabaseIntset(aclSelector *selector, int add_dbs) {
+    intset *new_dbs;
+
+    /* For removal operations with ALLDBS flag, convert to explicit list */
+    if (!add_dbs && (selector->flags & SELECTOR_FLAG_ALLDBS)) {
+        new_dbs = intsetNew();
+        if (!new_dbs) return NULL;
```

Comment:
Is this possible ? I don't think we perform this check anywhere. We would rather crash if we aren't able to allocate memory.